lxd/apparmor/archive.go

```diff
@@ -0,0 +1,146 @@
+package apparmor
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/lxc/lxd/lxd/sys"
+	"github.com/lxc/lxd/shared"
+)
+
+var archiveProfileTpl = template.Must(template.New("archiveProfile").Parse(`#include <tunables/global>
+profile "{{.name}}" {
+  /usr/lib/x86_64-linux-gnu/** mr,
+  /usr/lib/locale/locale-archive r,
+
+  {{ .rootPath }}/etc/ld.so.cache r,
+  {{ .rootPath }}/etc/locale.alias r,
+  {{ .rootPath }}/etc/nsswitch.conf r,
+  {{ .rootPath }}/etc/passwd r,
+  {{ .rootPath }}/etc/group r,
+
+  @{PROC}/filesystems r,
+  @{PROC}/@{pid}/mounts r,
+
+{{if .execPath -}}
+  {{ .execPath }} ux,
+{{- end }}
+{{if .unpackerPath -}}
+  {{ .unpackerPath }} ux,
+{{- end }}
+
+  {{ .outputPath }}/ rw,
+  {{ .outputPath }}/** rw,
+  {{ .varPath }}/backups/** rw,
```

Comment:
With owner prefix I've got this in tests:

`apparmor="DENIED" operation="open" profile="lxd_archive-home-piotr-go-src-github.com-lxc-lxd-test-tmp.3FI-sZg-storage-pools-lxdtest-sZg-custom-default_testvol" name="/home/piotr/go/src/github.com/lxc/lxd/test/tmp.3FI/sZg/storage-pools/lxdtest-sZg/custom/default_testvol/" pid=41557 comm="tar" requested_mask="r" denied_mask="r" fsuid=0 ouid=1000000`

Reply:
OK so `tar` runs as UID 1000000 and wants to read the storage pool dir owned by root. In that case, as you noticed `owner` won't work because the 2 don't align.